### PR TITLE
Use new docker image in all CircleCI steps

### PR DESCRIPTION
**Pull request by msimberg**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -5,8 +5,10 @@
 #  Distributed under the Boost Software License, Version 1.0. (See accompanying
 #  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
 
-defaults: &defaults
+working_dir_default: &working_dir_default
```
Comment (aurianer) [inline .circleci/config.yml:8]: Isn't it better to keep the target `defaults` depending on `working_dir_default` and `docker_default`?
```
defaults: &defaults
    <<: *docker_default
    <<: *working_dir_default
```

**Review by hkaiser — APPROVED** [review]
LGTM, thanks!

**Comment by aurianer** [comment]
Should we also update this one:
https://github.com/STEllAR-GROUP/hpx/pull/5226/files#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47R83 ?

**Comment by msimberg** [comment]
> Should we also update this one:
> https://github.com/STEllAR-GROUP/hpx/pull/5226/files#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47R83 ?

Definitely, good catch!

**Review by aurianer — APPROVED** [review]
LGTM thanks!

**Comment by msimberg** [comment]
@biddisco note that I've disabled the `scheduler_binding_check` on CircleCI as it was very consistently timing out. It may be that it just takes a long time to run (the timeout on CircleCI is now 120 seconds and the latest run I saw it pass it took 73 seconds). Would you mind having a look at it to see if it could be made to run a bit faster (possibly only in debug mode).

**Pull request merged**